python/mxnet/gluon/block.py

```diff
@@ -833,6 +848,12 @@ def _deferred_infer_shape(self, *args):
     def _call_cached_op(self, *args):
         if self._cached_op is None:
             self._build_cache(*args)
+        assert self._cached_op, "cached op is not None"
+        if self._callback:
+            self._cached_op._register_op_hook(self._callback, self._monitor_all)
+            if len(self._flags) >= 2 and (self._flags[1] or self._flags[0]):
```

Comment:
`self._flags = list(kwargs.items())`
The condition is true even when it is not supposed to be. For example if
`self._flags = [('static_alloc', False), ('static_shape', False), ('inline_limit', 2)]`